engines/scumm/smush/smush_player.cpp

```diff
@@ -1304,4 +1279,770 @@ void SmushPlayer::play(const char *filename, int32 speed, int32 offset, int32 st
+				if (chunk > _smushDispatch[idx].currentOffset) {
+					_smushTracks[idx].state = TRK_STATE_INACTIVE;
+					_smushTracks[idx].groupId = GRP_MASTER;
+					tmpFeedSize = 0;
+					break;
+				}
+
+				_smushDispatch[idx].audioRemaining += chunk;
+				_smushDispatch[idx].currentOffset -= chunk;
+			}
+			break;
+		case 7: // Set audio length
+			if (!_smushDispatch[idx].audioLength) {
+				_smushDispatch[idx].audioLength = READ_BE_UINT32(&code[6]);
+				_smushDispatch[idx].elapsedAudio = 0;
+			}
+
+			buf = _smushDispatch[idx].headerPtr;
+			_smushDispatch[idx].audioRemaining = _smushDispatch[idx].elapsedAudio + READ_BE_UINT32(buf + 2);
+
+			_smushDispatch[idx].currentOffset = READ_BE_UINT32(buf + 14);
+			if (_smushDispatch[idx].currentOffset > _smushDispatch[idx].audioLength)
+				_smushDispatch[idx].currentOffset = _smushDispatch[idx].audioLength;
+
+			_smushDispatch[idx].sampleRate = _smushAudioSampleRate;
+
+			_smushDispatch[idx].audioLength -= _smushDispatch[idx].currentOffset;
+			_smushDispatch[idx].elapsedAudio += _smushDispatch[idx].currentOffset;
+
+			if (_smushDispatch[idx].audioLength) {
+				_smushDispatch[idx].headerPtr = &code[code[1] + 2];
+			} else {
+				_smushDispatch[idx].headerPtr = &code[READ_BE_UINT16(&code[18])];
+			}
+
+			if (_smushDispatch[idx].audioRemaining >= _smushTracks[idx].availableSize + (_smushTracks[idx].availableSize >= _smushTracks[idx].sdatSize ? 0 : 15000) - _smushTracks[idx].dataSize) {
+				chunk = _smushTracks[idx].availableSize - _smushTracks[idx].dataSize - _smushDispatch[idx].audioRemaining + 15000;
+				if (chunk > _smushDispatch[idx].currentOffset) {
+					_smushTracks[idx].state = TRK_STATE_INACTIVE;
+					_smushTracks[idx].groupId = GRP_MASTER;
+					tmpFeedSize = 0;
+				} else {
+					_smushDispatch[idx].audioRemaining += chunk;
+					_smushDispatch[idx].currentOffset -= chunk;
+				}
+			}
+
+			break;
+		default:
+			_smushTracks[idx].state = TRK_STATE_INACTIVE;
+			_smushTracks[idx].groupId = GRP_MASTER;
+			tmpFeedSize = 0;
+		}
+
+		if (_smushDispatch[idx].currentOffset > 0) {
+			return false;
+		}
+	}
+
+	return true;
+}
+
+void SmushPlayer::sendAudioToDiMUSE(uint8 *mixBuf, int32 mixStartingPoint, int32 mixFeedSize, int32 mixInFrameCount, int volume, int pan) {
+	int clampedVol, clampedPan;
+	bool is11025Hz = false;
+
+	if (mixFeedSize == 2 * mixInFrameCount) {
+		is11025Hz = true;
+	} else if (mixFeedSize != mixInFrameCount) {
+		return;
+	}
+
+	clampedPan = CLIP<int>((pan / 2) + 64, 0, 127);
+	clampedVol = CLIP<int>(volume, 0, 127);
+	_imuseDigital->receiveAudioFromSMUSH(mixBuf, mixInFrameCount, mixFeedSize, mixStartingPoint, clampedVol, clampedPan, is11025Hz);
+}
+
+void SmushPlayer::feedAudio(uint8 *srcBuf, int groupId, int volume, int pan, int16 flags) {
+	int panDelta, effPan;
+	int32 maxFrames;
+	uint16 trkId, index;
+
+	if (_smushAudioInitialized) {
+		// Check file encoding
+		if (!srcBuf[8] && !srcBuf[9] && !srcBuf[12] && !srcBuf[13] && !srcBuf[16] && !srcBuf[17]) {
```

Comment:
preferable I would see "srcBuf[8] != 0" bytes are no boolean type, casting to Boolean.  this way is rather C thing, not C++

Reply:
Thanks, I didn't know that, I will correct that accordingly